src/contexts/videoListcontext.js

```diff
@@ -0,0 +1,50 @@
+import {createContext,useContext,useReducer} from "react";
+
+export const VideoPlayerContext=createContext();
+
+export const favoriteList=[];
+export const watchLaterList=[];
+
+ export function VideoPlayerContextProvider({children}){
+
+    const [state,dispatch]=useReducer(videoPlaylistReducer,{favoriteList,watchLaterList})
+    return (
+        <VideoPlayerContext.Provider value={{favoriteList:state.favoriteList,watchLaterList:state.watchLaterList,dispatch}}>{children}</VideoPlayerContext.Provider>
+    )
+}
+
+ export function useVideoPlayerContext(){
+    return useContext(VideoPlayerContext);
+}
+
+
+function videoPlaylistReducer(state,action){
+    switch(action.type){
+        case "ADD_TO_FAVORITES":
+            return{
+                favoriteList:state.favoriteList.concat(action.props.item),
+                watchLaterList: state.watchLaterList
+            };
+        break;
+        case "REMOVE_FROM_FAVORITES":
+            return{
+                favoriteList:state.favoriteList.filter((item)=>action.item.id !==item.id),
+                watchLaterList: state.watchLaterList
+            }
+            break;
+        case "ADD_TO_WATCHLATER":
+            return{
+                favoriteList:state.favoriteList,
```

Comment:
instead of writing favouriteList or watchLaterList if it is not changed, we can directly destructure the state ...state and make changes to required ones